theano/tensor/nnet/tests/test_abstract_conv.py

```diff
@@ -406,3 +410,270 @@ def test_grad_types(self):
+        kernel_2D = np.array([[1, 2, 3, 2, 1], [2, 4, 6, 4, 2],
+                              [3, 6, 9, 6, 3], [2, 4, 6, 4, 2],
+                              [1, 2, 3, 2, 1]])
+        np.testing.assert_allclose(kernel_2D, f())
+
+        # checking normalized filters
+        kernel = bilinear_kernel_2D(ratio=3, normalize=True)
+        f = theano.function([], kernel)
+        kernel_2D = kernel_2D / 9.
+        np.testing.assert_allclose(kernel_2D, f())
+
+        # checking kernel size 4
+        # checking un-normalized filters
+        kernel = bilinear_kernel_2D(ratio=4, normalize=False)
+        f = theano.function([], kernel)
+        kernel_2D = np.array([[1, 2, 3, 4, 3, 2, 1], [2, 4, 6, 8, 6, 4, 2],
+                              [3, 6, 9, 12, 9, 6, 3],
+                              [4, 8, 12, 16, 12, 8, 4],
+                              [3, 6, 9, 12, 9, 6, 3],
+                              [2, 4, 6, 8, 6, 4, 2], [1, 2, 3, 4, 3, 2, 1]])
+        np.testing.assert_allclose(kernel_2D, f())
+
+        # checking normalized filters
+        kernel = bilinear_kernel_2D(ratio=4, normalize=True)
+        f = theano.function([], kernel)
+        kernel_2D = kernel_2D / 16.
+        np.testing.assert_allclose(kernel_2D, f())
+
+    def test_bilinear_kernel_1D(self):
+        """Test 1D kernels used in bilinear upsampling
+
+        This method tests the correctness of the
+        1D kernel values used in bilinear upsampling
+        for some upsampling ratios.
+
+        """
+        # checking kernel size 2
+        kernel = bilinear_kernel_1D(ratio=2, normalize=False)
+        f = theano.function([], kernel)
+        kernel_1D = np.array([1, 2, 1])
+        np.testing.assert_allclose(kernel_1D, f())
+
+        # checking normalized filters
+        kernel = bilinear_kernel_1D(ratio=2, normalize=True)
+        f = theano.function([], kernel)
+        kernel_1D = kernel_1D / 2.
+        np.testing.assert_allclose(kernel_1D, f())
+
+        # checking kernel size 3
+        kernel = bilinear_kernel_1D(ratio=3, normalize=False)
+        f = theano.function([], kernel)
+        kernel_1D = np.array([1, 2, 3, 2, 1])
+        np.testing.assert_allclose(kernel_1D, f())
+
+        # checking normalized filters
+        kernel = bilinear_kernel_1D(ratio=3, normalize=True)
+        f = theano.function([], kernel)
+        kernel_1D = kernel_1D / 3.
+        np.testing.assert_allclose(kernel_1D, f())
+
+        # checking kernel size 4
+        kernel = bilinear_kernel_1D(ratio=4, normalize=False)
+        f = theano.function([], kernel)
+        kernel_1D = np.array([1, 2, 3, 4, 3, 2, 1])
+        np.testing.assert_allclose(kernel_1D, f())
+
+        # checking normalized filters
+        kernel = bilinear_kernel_1D(ratio=4, normalize=True)
+        f = theano.function([], kernel)
+        kernel_1D = kernel_1D / 4.
+        np.testing.assert_allclose(kernel_1D, f())
+
+    def make_1D_multiplier(self, ratio):
```

Comment:
The name of this function is not clear to me, `numerical_1D_kernel` seems to be more immediately comprehensible.